File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/quest/QuestObjectiveValidationMessage.scala





















// Generated on 05/03/2013 10:34:48
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait QuestObjectiveValidationMessage extends MessageDeserializer {
  val messageId = 6085

  protected trait Message extends BaseMessage with Serializable {
    val messageId = QuestObjectiveValidationMessage.messageId

    val questId: Short
    val objectiveId: Short

    def serialize(buf: Buffer) {
      buf.writeShort(questId)
      buf.writeShort(objectiveId)
    }

  }

  protected def create(questId: Short, objectiveId: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val questId = buf.readShort()
    val objectiveId = buf.readShort()

    create(questId, objectiveId)
  }

}

object QuestObjectiveValidationMessage extends QuestObjectiveValidationMessage {
  case class MessageImpl(questId: Short, objectiveId: Short) extends Message
  type Target = MessageImpl

  override def create(questId: Short, objectiveId: Short) = MessageImpl(questId, objectiveId)

  def apply(questId: Short, objectiveId: Short) = create(questId, objectiveId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/QuestObjectiveValidationMessage.java


// Generated on 05/08/2013 19:37:50
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class QuestObjectiveValidationMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6085;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short questId;
    public short objectiveId;
    
    public QuestObjectiveValidationMessage() { }
    
    public QuestObjectiveValidationMessage(short questId, short objectiveId) {
        this.questId = questId;
        this.objectiveId = objectiveId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(questId);
        buf.writeShort(objectiveId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        questId = buf.readShort();
        if (questId < 0)
            throw new RuntimeException("Forbidden value on questId = " + questId + ", it doesn't respect the following condition : questId < 0");
        objectiveId = buf.readShort();
        if (objectiveId < 0)
            throw new RuntimeException("Forbidden value on objectiveId = " + objectiveId + ", it doesn't respect the following condition : objectiveId < 0");
    }
    
}
